src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs

```diff
@@ -476,13 +476,36 @@ private SyntaxNode CreateThroughExpression(SyntaxGenerator factory)
                     // uncommon case and optimize for the common one - in other words, we only apply the cast
                     // in cases where we can unambiguously figure out which interface we are trying to implement.
                     var interfaceBeingImplemented = State.InterfaceTypes.SingleOrDefault();
-                    if ((interfaceBeingImplemented != null) && (!throughMemberType.Equals(interfaceBeingImplemented)))
+                    if ((interfaceBeingImplemented != null))
                     {
-                        through = factory.CastExpression(interfaceBeingImplemented,
-                            through.WithAdditionalAnnotations(Simplifier.Annotation));
+                        if (!throughMemberType.Equals(interfaceBeingImplemented))
+                        {
+                            through = factory.CastExpression(interfaceBeingImplemented,
+                                through.WithAdditionalAnnotations(Simplifier.Annotation));
+
+                            var facts = this.Document.GetLanguageService<ISyntaxFactsService>();
+                            through = facts.Parenthesize(through);
```

Comment:
`factory` is actually an instance of `SyntaxGenerator` not `SyntaxFactory`, which does not have a `ParenthesizedExpression` method.  I've renamed the local to clear up the confusion.  Is using facts service OK in that case?